### PR TITLE
:arrow_up: Upgrade `enum-as-inner` across crates

### PR DESCRIPTION
**Pull request by AucaCoyan**
WIP

### CODE REVIEW

**Comment by AucaCoyan** [comment]
Every other build is ready, but for some reason fails in windows. I don't understand the error 😢

**Comment by federico-terzi** [comment]
@AucaCoyan Thanks! I'm wondering if the windows error is caused by the Rust modules caching in the CI... Try to comment out this line: https://github.com/espanso/espanso/blob/dev/.github/workflows/ci.yml#L30

If it passes, we might need to update the Rust cache version :) https://github.com/Swatinem/rust-cache

**Comment by AucaCoyan** [comment]
Sadly it wasn't that,
Looking the files, i found that there are 2 dependencies that uses `windows` crate, one  "windows 0.51.1", and the other  "windows 0.24.0"
The old one is needed by `winrt-notification`, so I'll investigate if I can upgrade the windows crate to the same version

**Comment by AucaCoyan** [comment]
Aaaand nope, that is not it. Though it is still a better/newer dependency, so we can keep it

**Comment by AucaCoyan** [comment]
I rolled back the changes on the `espanso` crate and it worked. I see that we need to make this step first, and then in the forseable future update `enum-as-inner` in the future.
There is another PR already with this and another dependencies #1694 

If this PR gets landed after a review @federico-terzi , I would continue there!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-      - uses: Swatinem/rust-cache@v1
+      # - uses: Swatinem/rust-cache@v1
```
Comment (federico-terzi) [inline .github/workflows/ci.yml]: Disabling caching on the CI would bloat the times significantly (even up to 30 min at a time), can we revert it? :)

File: .gitignore [hunk 2]
```diff
@@ -37,4 +37,5 @@ DerivedData
 
 venv/
 
-.vscode/
\ No newline at end of file
+.vscode/
+*.profraw
```
Comment (federico-terzi) [inline .gitignore:41]: Why is this necessary? :)

**Review by federico-terzi — COMMENTED** [review]
Thanks! Left a couple of comments :)

File: .gitignore (hunk 2 above)
Comment (AucaCoyan) [reply .gitignore:41]: Because `.vscode/` is the folder where the Visual Studio Code saves the files of the editor. I think we shouldn't commit personal config files to the repo. Note that this prevents adding the folder .vscode when you `git add .`, but we can add a file with `--force` if needed (for example recommending and extension, or doing some build task, and so on)

`*.profraw` files are the result of doing coverage tests. [Elkiwa left a great guide on discord](https://discord.com/channels/884163483409731584/1013902234469269554/1186670417360977921)

**Comment by AucaCoyan** [comment]
Well, github actions failed because it didn't found the github server
```
Error: nodename nor servname provided, or not known (api.github.com:443)
```
Trying with another empty commit

**Comment by AucaCoyan** [comment]
Done! I also upgraded the rust edition from 2018 to 2021 😄

**Comment by federico-terzi** [comment]
Thanks! Looks great :)
